File: src/Shared/Logistics.Shared.Models/Trip/TripLoadDto.cs
using Logistics.Domain.Primitives.Enums;
using Logistics.Domain.Primitives.ValueObjects;

namespace Logistics.Shared.Models;

public record TripLoadDto
{
    public Guid Id { get; set; }
    public long Number { get; set; }
    public required string Name { get; set; }
    public LoadStatus Status { get; set; }
    public LoadType Type { get; set; }
    public double Distance { get; set; }
    public decimal DeliveryCost { get; set; }
    public required Address OriginAddress { get; set; }
    public required GeoPoint OriginLocation { get; set; }
    public required Address DestinationAddress { get; set; }
    public required GeoPoint DestinationLocation { get; set; }
    public required CustomerDto Customer { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/trip/trip-load.dto.ts
import {AddressDto} from "../address.dto";
import {CustomerDto} from "../customer";
import {GeoPointDto} from "../geo-point.dto";
import {LoadStatus, LoadType} from "../load";

export interface TripLoadDto {
  id: string;
  number: number;
  name: string;
  status: LoadStatus;
  type: LoadType;
  distance: number;
  deliveryCost: number;
  originAddress: AddressDto;
  originLocation: GeoPointDto;
  destinationAddress: AddressDto;
  destinationLocation: GeoPointDto;
  customer?: CustomerDto | null;
}
